mindmeld/auto_annotator.py

```diff
@@ -0,0 +1,708 @@
+        )
+        final_entities = self._resolve_conflicts(
+            current_entities=current_entities, annotated_entities=annotated_entities
+        )
+        processed_query.entities = tuple(final_entities)
+
+    def _get_annotated_entities(self, processed_query, entity_types=None):
+        """ Creates a list of query entities after parsing the text of a
+        processed query.
+
+        Args:
+            processed_query (ProcessedQuery): A processed query.
+            entity_types (list): List of entities allowed for annotation.
+
+        Returns:
+            query_entities (list): List of query entities.
+        """
+        if len(entity_types) == 0:
+            return []
+        entity_types = None if entity_types == ["*"] else entity_types
+        items = self.parse(
+            sentence=processed_query.query.text, entity_types=entity_types
+        )
+        query_entities = [
+            Annotator._item_to_query_entity(item, processed_query) for item in items
+        ]
+        return query_entities if len(query_entities) > 0 else []
+
+    @staticmethod
+    def _item_to_query_entity(item, processed_query):
+        """ Converts an item returned from parse into a query entity.
+
+        Args:
+            item (dict): Dictionary representing an entity with the keys -
+                "body", "start", "end", "value", "dim".
+            processed_query (ProcessedQuery): The processed query that the
+                entity is found in.
+
+        Returns:
+            query_entity (QueryEntity): The converted query entity.
+        """
+        span = Span(start=item["start"], end=item["end"] - 1)
+        entity = Entity(text=item["body"], entity_type=item["dim"], value=item["value"])
+        query_entity = QueryEntity.from_query(
+            query=processed_query.query, span=span, entity=entity
+        )
+        return query_entity
+
+    def _resolve_conflicts(self, current_entities, annotated_entities):
+        """ Resolve overlaps between existing entities and newly annotad entities.
+
+        Args:
+            current_entities (list): List of existing query entities.
+            annotated_entities (list): List of new query entities.
+
+        Returns:
+            final_entities (list): List of resolved query entities.
+        """
+        overwrite = self.config["overwrite"]
+        base_entities = annotated_entities if overwrite else current_entities
+        other_entities = current_entities if overwrite else annotated_entities
+
+        additional_entities = []
+        for o_entity in other_entities:
+            no_overlaps = [
+                Annotator._no_overlap(o_entity, b_entity) for b_entity in base_entities
+            ]
+            if all(no_overlaps):
+                additional_entities.append(o_entity)
+        return base_entities + additional_entities
+
+    @staticmethod
+    def _no_overlap(entity_one, entity_two):
```

Comment:
Lets add a test for this function

Reply:
To clarify, we want to add the _no_overlap() function to constants.py? Double checking since constants.py doesn't have functions yet